src/wp-includes/global-styles-and-settings.php

```diff
@@ -255,3 +287,14 @@ function ( $item ) {
 		}
 	}
 }
+
+/**
+ * Private function to clean the caches under the theme_json group.
+ *
+ * @since 6.2.0
+ * @access private
+ */
+function _wp_clean_theme_json_caches() {
+	wp_cache_delete( 'wp_get_global_settings', 'theme_json' );
```

Comment:
https://github.com/WordPress/wordpress-develop/pull/3556 https://github.com/WordPress/wordpress-develop/pull/3712 also clean caches. If any of them lands first, this line needs to be added to the existing method.

Reply:
I've updated your branch with latest `trunk` so this is good now.